android/app/src/main/java/com/tgmu/tgmu/ui/viewmodel/ExperienceViewModel.kt

```diff
@@ -1,59 +1,211 @@
 package com.tgmu.tgmu.ui.viewmodel
 
-import android.util.Log
+import android.net.Uri
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import androidx.lifecycle.map
 import androidx.lifecycle.viewModelScope
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
 import com.tgmu.tgmu.domain.model.Experience
+import com.tgmu.tgmu.domain.model.Movie
 import com.tgmu.tgmu.domain.repository.ExperienceRepository
 import com.tgmu.tgmu.utils.Resource
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.launch
+import java.util.Date
 import javax.inject.Inject
 
 @HiltViewModel
 class ExperienceViewModel @Inject constructor(private val experienceRepository: ExperienceRepository) :
     ViewModel() {
     private val _latestExperiences = MutableLiveData<Resource<List<Experience>>>()
-
     val latestExperiences: LiveData<Resource<List<Experience>>> get() = _latestExperiences
 
+    private val _specificExperience = MutableLiveData<Resource<Experience>>()
+
+    //    private val _selectedMovie = MutableLiveData<Movie>()
+//    private val _title = MutableLiveData<String>()
+//    private val _description = MutableLiveData<String>()
+    val specificExperience: LiveData<Resource<Experience>> get() = _specificExperience
+
+    private val _uploadStatus = MutableLiveData<Resource<Any>>()
+    val uploadStatus: LiveData<Resource<Any>> get() = _uploadStatus
+
+    //    val selectedMovie: LiveData<Movie> get() = _selectedMovie
+//    val title: LiveData<String> get() = _title
+//    val description: LiveData<String> get() = _description
+    private val _fileUploadURI = MutableLiveData<Uri>()
+    val fileUploadURI: LiveData<Uri> get() = _fileUploadURI
+
```

Comment:
Remove comments and think about implementing a data class of formData that you will work with it.